src/anm/analytics.js

```diff
@@ -0,0 +1,79 @@
+var engine = require('engine'),
+    utils = require('./utils.js');
+
+var Analytics = function () {
+    var self = this,
+        supportSendBeacon = !!navigator.sendBeacon,
+        timeout = 1000,
+        beacon = null,
+        animatronUrl = utils.makeApiUrl('/analytics/player');
+
+    self.enabled = animatronUrl != null && animatronUrl.indexOf('animatron-test') >= 0;
+    self.queue = [];
+
+    var event = function () {
+        if (self.queue.length > 0) {
+            var data = JSON.stringify(self.queue);
+            self.queue = [];
+
+            if (supportSendBeacon) {
+                navigator.sendBeacon(animatronUrl, data);
+            } else {
+                sendViaGif(data);
+            }
+        }
+        if (!supportSendBeacon) {
+            setTimeout(event, timeout);
+        }
+    };
+
+    var sendViaGif = function (data) {
+        if (!beacon) {
+            beacon = engine.createStatImg();
+        }
+        beacon.src = animatronUrl + '?data=' + encodeURIComponent(data);
+    };
+    if (self.enabled) {
+        if (!supportSendBeacon) {
+            engine.onDocReady(event);
```

Comment:
Sorry, but no :). _Which_ statistics exactly you want to send when document is ready? Or, ok, one second later. No player yet created. No scene yet loaded. And so no play or pause button yet pressed, no interactivity script performed. At all. For sure.

And there is no "initialization" code in `event` function, it only contains sending data or postponing of sending it for one or two damn seconds. I _may_ of course merge it, I just feel that for the sake of "safety" you do exactly what you call "premature optimization". 

I'd expect `event` call will be performed one single time on `window.unload`. No need in emptying the queue, just fill this queue with everything happened and send as one package. Or, use intervals instead. Or, for example, call it on `player.detach` for every player instance created. And I could be wrong, of course, but that's 30 lines of code and I still see no point in that. Sorry :(.